📖 Add documentation for nightly builds (#4968)
kubernetes-sigs/cluster-api-provider-aws: docs/book/src/development/nightlies.md

# Nightly Builds

Nightly builds are regular automated builds of the CAPA source code that occur every night.

> These builds are generated directly from the latest commit o source code on the main branch.

Reviewer: ```suggestion
These builds are generated directly from the latest commit of source code on the main branch.
```

I think my last suggestion had a typo :p

> These builds are generated directly from the latest version of the source code repository, the main branch.

Reviewer: ```suggestion
These builds are generated directly from the latest commit o source code on the main branch.
```
